### ISSUE TITLE
Cannot restore config value from store view scope using "Use website" checkbox.

### ISSUE THREAD

**Issue by brian-ebizmarts** [magento/magento2#26732]
### Preconditions (*)

1. Magento >=2.2

### Steps to reproduce (*)

1. Uncheck "Use website" and set specific setting at store view level, save config. 
2. Check "Use website" and save config.

### Expected result (*)

1. delete value from table core_config_data

### Actual result (*)

1. Cannot inherit values from parent scopes. 

### Additional information

I found in xdebug that the function **getFieldPath** on the **module-config** is called with wrong parameters.

Function **getFieldPath** is declared as below [Model/Config.php#L294](https://github.com/magento/magento2/blob/2.3/app/code/Magento/Config/Model/Config.php#L294)

https://github.com/magento/magento2/blob/5f3b86ab4bd3e3b94e65429fed33f748d29c1bbe/app/code/Magento/Config/Model/Config.php#L285-L294

In function **_processGroup** the parameter $extraOldGroups is passed instead of $oldConfig. 
[Config/Model/Config.php#L456](https://github.com/magento/magento2/blob/2.3/app/code/Magento/Config/Model/Config.php#L456)

https://github.com/magento/magento2/blob/5f3b86ab4bd3e3b94e65429fed33f748d29c1bbe/app/code/Magento/Config/Model/Config.php#L456

As these two parameters are passed by reference $extraOldGroups overwrite the $oldConfig consequently the checkbox status won't validate and the value is not deleted.

https://github.com/magento/magento2/blob/5f3b86ab4bd3e3b94e65429fed33f748d29c1bbe/app/code/Magento/Config/Model/Config.php#L460-L471

I've noticed inheritance problems between scopes when trying to use multiples sections in system.xml.

**Comment by gauravagarwal1001** [comment]
@magento give me 2.4-develop instance

**Comment by magento-engcom-team** [comment]
Hi @gauravagarwal1001. Thank you for your request. I'm working on Magento 2.4-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @gauravagarwal1001, here is your [Magento instance](https://i-26732-2-4-develop.instances.magento-community.engineering).
Admin access: [https://i-26732-2-4-develop.instances.magento-community.engineering/admin_e554](https://i-26732-2-4-develop.instances.magento-community.engineering/admin_e554)
Login: `5c1a94c8` Password: `01f7f3143cb0`
Instance will be terminated in up to 3 hours.

**Comment by shikhamis11** [comment]
Hi @brian-ebizmarts 
I checked this issue is not reproducible at 2.4-develop instance .. let me know if it requires any additional steps to reproduce this issue .. also please share video or screenshots if you are able to reproduce this issue .

**Comment by brian-ebizmarts** [comment]
Hi @shikhamis11, here is a video in Magento EE 2.3.2 instance:
[https://drive.google.com/file/d/1UUa1HlARvJ83NOH1FgLY_AlcsMZ3XCvG/view?usp=sharing](https://drive.google.com/file/d/1UUa1HlARvJ83NOH1FgLY_AlcsMZ3XCvG/view?usp=sharing)

Let me know if you need anything else.

**Comment by brian-ebizmarts** [comment]
Hi @shikhamis11, this issue is reproducible with multiples sections in the System.xml:

```
<?xml version="1.0"?>
<config xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="urn:magento:framework:Magento/Config/etc/system_file.xsd">
    <system>
        <tab id="pos" translate="label" sortOrder="550">
            <label>POS by Ebizmarts</label>
        </tab>
        <section id="pos_general" translate="label" type="text" sortOrder="100" showInDefault="1" showInWebsite="1" showInStore="1">
           .....
        </section>
        <section id="pos_payment" translate="label" type="text" sortOrder="200" showInDefault="1" showInWebsite="1" showInStore="1">
            <class>separator-top</class>
            <label>Payment Methods</label>
            <tab>pos</tab>
            <resource>Ebizmarts_Pos::pos_config</resource>
            .....
        </section>
        <section id="pos_shipping" translate="label" type="text" sortOrder="300" showInDefault="1" showInWebsite="1" showInStore="1">
            <label>Shipping methods</label>
            <tab>pos</tab>
            <resource>Ebizmarts_Pos::pos_config</resource>
            .....
        </section>
        <section id="pos_cron" translate="label" type="text" sortOrder="400" showInDefault="1" showInWebsite="1" showInStore="1">
            <label>Cron</label>
            <tab>pos</tab>
            <resource>Ebizmarts_Pos::cron_pos</resource>
            .....
        </section>
        <section id="pos_backup" translate="label" type="text" sortOrder="500" showInDefault="1" showInWebsite="1" showInStore="1">
            <label>Backups</label>
            <tab>pos</tab>
            <resource>Ebizmarts_Pos::backup_pos</resource>
            .....
        </section>
    </system>
</config>
```
I've to mention that only in the first section id(pos_general) the checkbox "Use website" works as expected and delete the entry in core_config_data table. In the other sections pos_payment, pos_shipping, pos_cron, pos_backup is not working. 

If I replace this line:
https://github.com/magento/magento2/blob/5f3b86ab4bd3e3b94e65429fed33f748d29c1bbe/app/code/Magento/Config/Model/Config.php#L456

To pass the parameters as declared it works correctly:
`$path = $this->getFieldPath($field, $fieldId, $oldConfig, $extraOldGroups);`

**Comment by engcom-Bravo** [comment]
Hello @brian-ebizmarts 

Thank you for your report.

It looks like you have third-party extensions installed on your Magento instance.
Unfortunately, we have to close this issue. The GitHub issue tracker is intended for Magento Core technical issues only.
Please refer to the [Community Forums](https://community.magento.com/) or the [Magento Stack Exchange](https://magento.stackexchange.com/) site for advice or general discussion about this issue. 

Thank you

**Comment by mam08ixo** [comment]
The bug is **revealed** by a third-party extension, yes. But it is still **caused** by a Magento Core technical issue.

I can replicate it by setting a field's `config_path` property to a different section. Simplified example:

```xml
<section id="foo">
    <group id="b">
        <field id="c" showInDefault="1" showInWebsite="1" showInStore="1">
            <label>Foo</label>
            <config_path>bar/b/c</config_path>
        </field>
    </group>
</section>
```

Fixing the argument order as detected by @brian-ebizmarts solves the issue.

@shikhamis11 @engcom-Bravo This needs to be reopened.

**Comment by sdzhepa** [comment]
Based on feedback it seems issue should be reopened and re-verified one more time on the `2.4-develop`

**Comment by engcom-Alfa** [comment]
HI @brian-ebizmarts . Thank you for your report!
Unfortunately, we are not able to reproduce this issue on fresh 2.4-develop. Everything works as expected.
Looks like the problem has already been fixed by commit 005b10d7cd.

@brian-ebizmarts Could you take a look?
Thanks!

**Comment by brian-ebizmarts** [comment]
Hi @engcom-Alfa, 

The problem has been solved by the commit 005b10d. Would it be possible to add this fix on **Magento 2.3**?

Thanks in advance.

**Comment by engcom-Alfa** [comment]
Hi @brian-ebizmarts . 

Unfortunately, there is no such possibility to add this fix on Magento 2.3.
According to the [DevBlog](https://community.magento.com/t5/Magento-DevBlog/Let-s-Keep-the-Magento-2-Backlog-Clean-Part-2/ba-p/462599):

`
The latest Magento version as of now is 2.4.1 and moving forward we will be focusing on the 2.4 and 2.5 release lines. As a result of such an approach, we will be parking the delivery of bug fixes or features for the 2.3 line and only delivering security fixes to it.
`

Thanks!

**Comment by engcom-Alfa** [comment]
Will close this issue as the problem has already been fixed by commit 005b10d.
